apiserver/facades/client/application/application.go

```diff
@@ -2557,20 +2557,14 @@ func validateAgentVersions(
 		return errors.Trace(err)
 	}
 	if errors.Is(err, errors.NotFound) || ver.Compare(epoch) >= 0 {
-		// Check to see if the model config version is valid
-		// Arguably we could check on the per-unit level, as that is the
-		// *actual* version of the agent that is running, looking at the
-		// versioner (alias to model config), we get the intent of the move
-		// to that version.
-		// This should be enough for a pre-flight check, rather than querying
-		// potentially thousands of units (think large production stacks).
-		modelVer, modelErr := modelAgentService.GetModelAgentVersion(ctx)
-		if modelErr != nil {
-			// If we can't find the model config version, then we can't do the
-			// comparison check.
-			return errors.Trace(modelErr)
-		}
-		if modelVer.Compare(epoch) < 0 {
+		appAgentVer, err := modelAgentService.GetApplicationTargetAgentVersion(
```

Comment:
It looks like we can remove the validateAgentVersions method and where it's called as it's checking for a version less than juju 2.6 which is impossible with a 4.0 controller.  If this is the only use GetApplicationTargetAgentVersion, that can go too.

Reply:
All blown away 🎉